app/knight.py

```diff
@@ -0,0 +1,45 @@
+class Knight:
+
+    knights_instances = {}
+
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 weapon: dict,
+                 armour: list = [],
+                 potion: dict = {}) -> None:

```

Comment:
Using a mutable default argument like a dictionary for `potion` can lead to unexpected behavior. Consider using `None` as the default value and initializing it inside the constructor if `None` is passed.